src/itree.cpp

```diff
@@ -0,0 +1,133 @@
+#include "rodsClient.h"
+#include "client_connection.hpp"
+#include <boost/program_options.hpp>
+#include "irods_client_api_table.hpp"
+#include "irods_pack_table.hpp"
+#include "irods_parse_command_line_options.hpp"
+#include "irods/filesystem.hpp"
+
+#include "fmt/core.h"
+#include "fmt/color.h"
+
+namespace po = boost::program_options;
+namespace fs = irods::experimental::filesystem;
+auto print_dir(const fs::path&,
+               const po::variables_map&,
+               irods::experimental::client_connection&) -> void;
+
+auto correct_path(const po::variables_map&,const rodsEnv&)->fs::path;
+
+// ANSI escape numbers. For setting the colors.
+static const auto directory_color=fmt::color::blue;
+static const auto file_color = fmt::color::green;
+
+int main(int argc, char** argv){
+    rodsEnv env;
+    int status = getRodsEnv(&env);
+    irods::experimental::client_connection conn;
+    po::options_description desc{""};
+    po::positional_options_description pod;
+    pod.add("dir",1);
+    desc.add_options()
+        ("help","Display help")
+        ("dir",po::value<std::string>()->default_value(env.rodsCwd), "The collection to start from(defaults to the current working collection)")
+        ("directories_only,d", po::bool_switch(),"Only list directories")
```

Comment:
hrm, feel strongly both ways about this one... `d` b/c consistency with unix tools... but we have Collections...

`-c` already would collide with `color`... `-C` too subtle?  thoughts anyone?

Reply:
Maybe `--color` loses its short option. `--collections-only` steals `-c`. Another solution could be that `--collections-only` doesn't get a short option. However, we define the options and there is no reason we can't use some other letter.

Also, to be more like Unix/Linux, use hyphens for long option names (to separate words).

Reply:
`-C` seems like a good choice for this, it also happens to be the switch that `ls` uses to control coloring, so it's more consistent anyway.

I could also make it accept `-c` in addition to `-d`